### DISCUSSION TITLE
Bug when retrieving sprints?

### DISCUSSION THREAD

**Discussion by ghost** [pycontribs/jira#1280] [Q&A]
It seems I've run into a bug and would like to verify if others run into the same problem?

When requesting all the sprints for a board the total value provided by the return list, seems to be always equal to the maxResults value provided.

If I use:
```
        all_sprints: ResultList[Sprints] = jira.sprints(
            board_id=board_id,
            startAt=num_sprints,
            # maxResults=page_size
        )
```
Then all_sprints.total has a value of 50.
If I use maxResults=1, then all_sprint.total has a value of 1.

I'm using Jira 3.1.1, python 3.8.10 on Xubuntu, addressing Jira Cloud.

Edits:
When using jira.boards(...) this is not the case and everything works as expected.

The options used:
```
{'server': 'XXX, 'auth_url': '/rest/auth/1/session', 'context_path': '/', 'rest_path': 'api', 'rest_api_version': '2', 'agile_rest_path': 'agile', 'agile_rest_api_version': '1.0', 'verify': True, 'resilient': True, 'async': False, 'async_workers': 5, 'client_cert': None, 'check_update': False, 'delay_reload': 0, 'headers': {'Cache-Control': 'no-cache', 'Content-Type': 'application/json', 'X-Atlassian-Token': 'no-check'}
```

**Comment by ghosharindam** [comment]
I have also run into a similar issue. Setting maxResults>50 doesn't have any effect. the maximum number of sprints returned is still the first 50 sprints. This makes it kind of unusable - since the latest sprints are the most important. Setting maxResults < 50, returns the first n sprints where n <=50. Is this the intended behaviour and how could I get all the sprints?

**Comment by studioj** [comment]
An issue was reported stating the same behavior...
https://github.com/pycontribs/jira/issues/932
If I find the time I could have a look. No one is stopping you from having a look yourself, we're open for contributions.

Kind regards

**Reply by studioj** [reply]
Thanks for the info @CoenSchalkwijk and @ghosharindam
